pkg/broker/api_test.go

```diff
@@ -108,11 +108,46 @@ func (db mockDataStoreProvision) GetServiceBinding(id string) (*serviceinstance.
 	switch id {
 	case "err":
 		return nil, errors.New("test failure")
+	case "err-instance":
```

Comment:
This was the simplest way to set up the different test cases, but I think it'd be cleaner long-term to make the changes required to use a `dynamodbiface.DynamoDBAPI` instead of the `mockDataStoreProvision`. Something for a future PR.

Reply:
The thought behind the extra layer of abstraction with the datastore interface was to make it easy to implement alternative persistent storage options like etcd. Not sure if it really does make it that much easier, or if it's something that we'll ever do, so I'm not opposed to getting rid of it.